README.md

```diff
@@ -1,11 +1,65 @@
+2. On-demand account loading. If a AWS account is not found in the local repository at the time of pipeline execution, 
+the plugin will perform a sync with remote host to provide needed account (if found in remote host).
+3. Supports IAM authentication when used with API Gateway. The Spinnaker managing account role must have the permission to invoke configured API gateway.   
 
 ### Requirements
-1. Must be used with Spinnaker version 1.22 or higher.
-2. Must enable [Lambda support](https://kb.armory.io/s/article/AWS-Lambda-Custom-Webhook-Stages).
-3. Must enable [ECS support](https://spinnaker.io/setup/install/providers/aws/aws-ecs/#clouddriver-yaml-properties)
+1. Must be used with Spinnaker version 1.24 or higher.
+2. Must enable [AWS support](https://docs.armory.io/docs/armory-admin/aws/add-aws-account/)
+3. Must enable [Lambda support](https://kb.armory.io/s/article/AWS-Lambda-Custom-Webhook-Stages).
+4. Must enable [ECS support](https://spinnaker.io/setup/install/providers/aws/aws-ecs/#clouddriver-yaml-properties)
+5. Must have a HTTP endpoint which provides [JSON payload](#Expected-JSON-payload) when invoked with `GET` (Example available at `.github/integration-testing/mock-server/`)
+
+### Expected JSON payload
+This plugin expects the following JSON payload from the configured remote host, configured with the `url` property.
+
+
+```json
+{
+  "SpinnakerAccounts": [
+    {
+      "AccountId": "12345678901",
+      "SpinnakerAccountName": "test-3",
+      "Regions": [
+        "us-west-2"
+      ],
+      "SpinnakerStatus": "ACTIVE | SUSPENDED",
+      "SpinnakerAssumeRole": "role/spinnakerManaged",
+      "SpinnakerProviders": [
+        "ecs", "lambda", "ec2"
+      ],
+      "UpdatedAt": "2020-08-27T16:52:59.026696+00:00"
+    }
+  ],
+  "Pagination": {
+    "NextUrl": "http://some/next/url"
+  }
+}
+```
+
+## Note
+1. Plugin performs `GET` with query string field `UpdatedAt.gt=<TIME>` after the initial sync.
+Expectation is that the remote host will return accounts that were updated after the specified time by the field.
+This is done to avoid returning and processing all accounts every time sync occurs. 
+2. The `UpdatedAt.gt` field value is determined using the most recent time value provided in the `UpdatedAt` JSON field.
+E.g. if two accounts were retruned with timestamps `2020-08-27T16:52:59.026696+00:00` and `2030-12-27T16:52:59.026696+00:00`, 
+next request will have a query string field `UpdatedAt.gt=2030-12-27T16:52:59.026696+00:00`. 
+2. The `SpinnakerProviders` JSON field means the following:
+    - If empty, AWS and ECS accounts are removed.
+    - If only `ec2` is specified, Spinnaker AWS account is created.
+    - If only `lambda` is specified, Spinnaker AWS account with Lambda support is created.
+    - If `ecs` is specified, Spinnaker AWS and ECS accounts are created.
+3. ECS accounts are named with corresponding AWS account's name with "-ecs" suffix. 
```

Comment:
this is something we can probably look a bit more into. The naming for the accounts doesnt need to be unique across providers. Do we still need to make this adjustment to the names created by the plugin?

Reply:
Good point. Previously it was necessary to have unique names. I think this is something that can be worked on in the future.